R/tostring.R

```diff
@@ -1,43 +1,52 @@
-## this can't be tested from within R
-# nocov start
-#' @importFrom stats na.omit
-#' @importFrom utils head tail localeToCharset
-#' @import checkmate
-
-d_hsep_factory <- function() {
-  warn_sent <- FALSE
-  function() {
-    if (any(grepl("^UTF", localeToCharset()))) {
-      "\u2014"
-    } else {
-      if (!warn_sent && interactive()) {
-        message(
-          "Detected non-UTF charset. Falling back to '-' ",
-          "as default header/body separator. This warning ",
-          "will only be shown once per R session."
-        )
-        warn_sent <<- TRUE
-      }
-      "-"
-    }
-  }
-}
-
-#' Default horizontal Separator
+#' @title Default horizontal Separator
 #'
-#' The default horizontal separator character which can be
+#' @description The default horizontal separator character which can be
 #' displayed in the current `charset` for use in rendering table-likes.
 #'
+#' @param hsep_chr character(1). Character that will be set in the R environment
```

Comment:
Maybe change the name of the param to `hsep_char` for better understandability? 

```suggestion
#' @param hsep_char character(1). Character that will be set in the R environment
```